zcash-issue-dag.py

```diff
@@ -308,7 +330,19 @@ def main():
             dg.remove_nodes_from(nx.compose_all(ignore))
 
     # Prune nodes that are not downstream of any open issues.
-    if PRUNE_FINISHED:
+    if PRUNE_FINISHED == 'targets':
+        closed_targets = [n for n in dg.nodes if n.is_target and n.state == 'closed']
+        for target in closed_targets:
+            try:
+                ancestors = nx.ancestors(dg, target)
+                if not any(n.state != 'closed' for n in ancestors):
+                    # Only prune ancestors, not the closed target node, so that
+                    # we see the most recently-closed target nodes in the DAG.
+                    dg.remove_nodes_from(ancestors)
+            except nx.NetworkXError:
+                # One of the ancestors was a previously-removed target.
```

Comment:
```suggestion
            if target in dg:
                ancestors = nx.ancestors(dg, target)
                if all(n.state == 'closed' for n in ancestors):
                    # Only prune ancestors, not the closed target node, so that
                    # we see the most recently-closed target nodes in the DAG.
                    dg.remove_nodes_from(ancestors)
```